framework/src/org/checkerframework/common/signatureinference/SignatureInferenceScenes.java

```diff
@@ -0,0 +1,588 @@
+            return BasicAFT.forType(float.class);
+        else if (value instanceof Integer)
+            return BasicAFT.forType(int.class);
+        else if (value instanceof Long)
+            return BasicAFT.forType(long.class);
+        else if (value instanceof Short)
+            return BasicAFT.forType(short.class);
+        else if (value instanceof String)
+            return BasicAFT.forType(String.class);
+        return null;
+    }
+
+
+    /**
+     *  Adds a field to an AnnotationBuilder.
+     * @param fieldKey is the name of the field
+     * @param obj is the value of the field
+     * @param builder is the AnnotationBuilder
+     */
+    @SuppressWarnings("unchecked") // This is actually checked in the first
+    //instanceOf call.
+    private static void addFieldToAnnotationBuilder(String fieldKey, Object obj,
+            AnnotationBuilder builder) {
+        if (obj instanceof List<?>) {
+            builder.setValue(fieldKey, (List<Object>) obj);
+        } else if (obj instanceof String) {
+            builder.setValue(fieldKey, (String)obj);
+        } else if (obj instanceof Integer) {
+            builder.setValue(fieldKey, (Integer)obj);
+        } else if (obj instanceof Float) {
+            builder.setValue(fieldKey, (Float)obj);
+        } else if (obj instanceof Long) {
+            builder.setValue(fieldKey, (Long)obj);
+        } else if (obj instanceof Boolean) {
+            builder.setValue(fieldKey, (Boolean)obj);
+        } else if (obj instanceof Character) {
+            builder.setValue(fieldKey, (Character)obj);
+        } else if (obj instanceof Class<?>) {
+            builder.setValue(fieldKey, (Class<?>)obj);
+        } else if (obj instanceof Double) {
+            builder.setValue(fieldKey, (Double)obj);
+        } else if (obj instanceof Float) {
+            builder.setValue(fieldKey, (Float)obj);
+        } else if (obj instanceof Enum<?>) {
+            builder.setValue(fieldKey, (Enum<?>)obj);
+        } else if (obj instanceof Enum<?>[]) {
+            builder.setValue(fieldKey, (Enum<?>[])obj);
+        } else if (obj instanceof AnnotationMirror) {
+            builder.setValue(fieldKey, (AnnotationMirror)obj);
+        } else if (obj instanceof Object[]) {
+            builder.setValue(fieldKey, (Object[])obj);
+        } else if (obj instanceof TypeMirror) {
+            builder.setValue(fieldKey, (TypeMirror)obj);
+        } else if (obj instanceof Short) {
+            builder.setValue(fieldKey, (Short)obj);
+        } else if (obj instanceof VariableElement) {
+            builder.setValue(fieldKey, (VariableElement)obj);
+        } else if (obj instanceof VariableElement[]) {
+            builder.setValue(fieldKey, (VariableElement[])obj);
+        } else {
+            ErrorReporter.errorAbort("Unrecognized type: " + obj.getClass());
+        }
+    }
+
+    /**
+     * Returns the ClassSymbol of the class encapsulating
+     * the node n passed as parameter.
```

Comment:
How are classTree and receiverNode used?  The specification doesn't mention them.